src/Core/Managed/Shared/TelemetryClientExtensions.cs

```diff
@@ -93,26 +94,78 @@ public static IOperationHolder<T> StartOperation<T>(this TelemetryClient telemet
                 operationTelemetry.GenerateOperationId();
             }
 
+            var telemetryContext = operationTelemetry.Context.Operation;
+
             // If the operation is not executing in the context of any other operation
             // set its name and id as a context (root) operation name and id
-            if (string.IsNullOrEmpty(operationTelemetry.Context.Operation.Id))
+            if (string.IsNullOrEmpty(telemetryContext.Id))
             {
-                operationTelemetry.Context.Operation.Id = operationTelemetry.Id;
+                telemetryContext.Id = operationTelemetry.Id;
             }
 
-            if (string.IsNullOrEmpty(operationTelemetry.Context.Operation.Name))
+            if (string.IsNullOrEmpty(telemetryContext.Name))
             {
-                operationTelemetry.Context.Operation.Name = operationTelemetry.Name;
+                telemetryContext.Name = operationTelemetry.Name;
             }
 
+            bool isActivityEnabled = false;
+#if !NET40
+            if (ActivityExtensions.IsActivityEnabled())
+            {
+                bool operationNameIsSet = false;
+
+                var childActivity = new Activity("Microsoft.ApplicationInsights.OperationContext");
```

Comment:
It makes sense to make it const, but in the StopOperation we check for activity.Id which is globally unique, so there is no reason to check for Activity name. 

I also want to stop checking (AI) operation name: it's redundant and could cause telemetry being lost if operation name changes